pkg/orchestrator/applications_gateway.go

```diff
@@ -50,24 +50,45 @@ func (gateway ApplicationsDataGateway) Find() ([]ApplicationRecord, error) {
 	return records, nil
 }
 
-func (gateway ApplicationsDataGateway) FindByIntegrationId(integrationId string) (ApplicationRecord, error) {
+func (gateway ApplicationsDataGateway) FindByIntegrationId(integrationId string) (*ApplicationRecord, error) {
 	s := "select id, integration_id, object_id, name, description, service from applications where integration_id=$1"
 	return gateway.queryRow(s, integrationId)
 }
 
-func (gateway ApplicationsDataGateway) FindByObjectId(objectId string) (ApplicationRecord, error) {
+func (gateway ApplicationsDataGateway) FindByObjectId(objectId string) (*ApplicationRecord, error) {
 	s := "select id, integration_id, object_id, name, description, service from applications where object_id=$1"
-	return gateway.queryRow(s, objectId)
+	return gateway.query(objectId, s)
 }
 
-func (gateway ApplicationsDataGateway) FindById(id string) (ApplicationRecord, error) {
+func (gateway ApplicationsDataGateway) FindById(id string) (*ApplicationRecord, error) {
 	s := "select id, integration_id, object_id, name, description, service from applications where id=$1"
 	return gateway.queryRow(s, id)
 }
 
-func (gateway ApplicationsDataGateway) queryRow(sql string, id string) (ApplicationRecord, error) {
+func (gateway ApplicationsDataGateway) DeleteById(id string) error {
```

Comment:
I was thinking of making this accept variadic string ids. `DeleteById(id ...string)`. 

If we passed in a single `id`, we'd run  the query `delete from applications where id`. But if we passed in multiple `id`s then we could run the query `delete from applications where id=any($1)`.

This has the benefit of allowing a single delete sql operation on multiple ids. But it does add complexity to this method and its usage since it technically runs different queries depending on the number of arguments passed in.